This patch fixes several things:
 - the package installs per default binaries with dangling libraries deps
   (installs the binaries without the required libraries)
 - libmfscommon must be shared to link against libfmt
 - any library which isn't indirected via shared_add_library should be STATIC
 - rename libmount to avoid install-time conflict with util-linux' libmount
 - rename libcrcutil, just to be sure
 - remove the partially duplicated logic of shared_add_library from src/mount/client
diff --git a/cmake/CreateUnitTest.cmake b/cmake/CreateUnitTest.cmake
index ac9a87e..8588656 100644
--- a/cmake/CreateUnitTest.cmake
+++ b/cmake/CreateUnitTest.cmake
@@ -23,7 +23,7 @@ function(create_unittest TEST_NAME)
 
   include_directories(${GTEST_INCLUDE_DIRS})
 
-  add_library(${TEST_LIBRARY_NAME} ${ARGV})
+  add_library(${TEST_LIBRARY_NAME} STATIC ${ARGV})
 
   list(FIND UNITTEST_TEST_NAMES ${TEST_NAME} result)
   if (${result} EQUAL -1)
diff --git a/cmake/SharedLibraries.cmake b/cmake/SharedLibraries.cmake
index 631c1a3..63a50ec 100644
--- a/cmake/SharedLibraries.cmake
+++ b/cmake/SharedLibraries.cmake
@@ -16,16 +16,8 @@
 
 function(shared_add_library NAME ...)
 	list(REMOVE_AT ARGV 0)
-
-	add_library(${NAME} ${ARGV})
-	if(ENABLE_PIC_TARGETS)
-		add_library(${NAME}_pic ${ARGV})
-		if("${CMAKE_VERSION}" VERSION_GREATER 2.8.9)
-			set_property(TARGET ${NAME}_pic PROPERTY POSITION_INDEPENDENT_CODE ON)
-		else()
-			set_property(TARGET ${NAME}_pic PROPERTY COMPILE_FLAGS "-fPIC")
-		endif()
-	endif()
+	add_library(${NAME} SHARED ${ARGV})
+	install(TARGETS ${NAME} DESTINATION RUNTIME DESTINATION ${LIB_SUBDIR})
 endfunction()
 
 function(shared_target_link_libraries TARGET)
diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 0fd9f46..661a2a4 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -6,5 +6,6 @@ if(NOT CRCUTIL_FOUND)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CRCUTIL_CXX_FLAGS}")
     file(GLOB CRCUTIL_SOURCES ${CRCUTIL_SOURCE_DIR}/*.cc)
     shared_add_library(crcutil ${CRCUTIL_SOURCES})
+    set_target_properties(crcutil PROPERTIES OUTPUT_NAME "mfs-crcutil")
   endif()
 endif()
diff --git a/src/admin/CMakeLists.txt b/src/admin/CMakeLists.txt
index ec7b218..7a361d4 100644
--- a/src/admin/CMakeLists.txt
+++ b/src/admin/CMakeLists.txt
@@ -1,6 +1,6 @@
 collect_sources(LIZARDFS_ADMIN)
 
-add_library(lizardfs-admin-lib ${LIZARDFS_ADMIN_SOURCES})
+add_library(lizardfs-admin-lib STATIC ${LIZARDFS_ADMIN_SOURCES})
 target_link_libraries(lizardfs-admin-lib mfscommon)
 create_unittest(lizardfs-admin-lib ${LIZARDFS_ADMIN_TESTS})
 link_unittest(lizardfs-admin-lib mfscommon)
diff --git a/src/chunkserver/CMakeLists.txt b/src/chunkserver/CMakeLists.txt
index 9c38a79..76380d1 100755
--- a/src/chunkserver/CMakeLists.txt
+++ b/src/chunkserver/CMakeLists.txt
@@ -3,7 +3,7 @@ add_definitions(-DLIZARDFS_MAX_FILES=10000)
 add_definitions(-DAPPNAME=mfschunkserver)
 
 collect_sources(CHUNKSERVER)
-add_library(chunkserver ${CHUNKSERVER_SOURCES})
+add_library(chunkserver STATIC ${CHUNKSERVER_SOURCES})
 target_link_libraries(chunkserver lzfsprotocol mfscommon ${ADDITIONAL_LIBS})
 create_unittest(chunkserver ${CHUNKSERVER_TESTS})
 link_unittest(chunkserver chunkserver mfscommon)
diff --git a/src/devtools/CMakeLists.txt b/src/devtools/CMakeLists.txt
index 2d9a827..ff72126 100644
--- a/src/devtools/CMakeLists.txt
+++ b/src/devtools/CMakeLists.txt
@@ -1,4 +1,4 @@
 aux_source_directory(${CMAKE_CURRENT_SOURCE_DIR} DEVTOOLS_SOURCES)
-add_library(devtools ${DEVTOOLS_SOURCES})
+add_library(devtools STATIC ${DEVTOOLS_SOURCES})
 
 add_subdirectory(mycrc32)
diff --git a/src/master/CMakeLists.txt b/src/master/CMakeLists.txt
index 0a35a6c..fbd3f53 100755
--- a/src/master/CMakeLists.txt
+++ b/src/master/CMakeLists.txt
@@ -7,7 +7,7 @@ if(NOT DB_FOUND)
   list(REMOVE_ITEM MASTER_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/hstring_bdbstorage.cc")
 endif()
 
-add_library(master ${MASTER_SOURCES})
+add_library(master STATIC ${MASTER_SOURCES})
 target_link_libraries(master mfscommon ${ADDITIONAL_LIBS})
 if(JUDY_LIBRARY)
   target_link_libraries(master ${JUDY_LIBRARY})
diff --git a/src/metadump/CMakeLists.txt b/src/metadump/CMakeLists.txt
index 3ef93f6..0db19ea 100755
--- a/src/metadump/CMakeLists.txt
+++ b/src/metadump/CMakeLists.txt
@@ -2,5 +2,4 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 aux_source_directory(. METADUMP_SOURCES)
 add_executable(mfsmetadump ${METADUMP_SOURCES})
-target_link_libraries(mfsmetadump)
 install(TARGETS mfsmetadump RUNTIME DESTINATION ${SBIN_SUBDIR})
diff --git a/src/metalogger/CMakeLists.txt b/src/metalogger/CMakeLists.txt
index cbd75be..ed0d995 100755
--- a/src/metalogger/CMakeLists.txt
+++ b/src/metalogger/CMakeLists.txt
@@ -4,7 +4,7 @@ add_definitions(-DAPPNAME=mfsmetalogger)
 
 collect_sources(METALOGGER)
 
-add_library(metalogger ${METALOGGER_SOURCES} ../master/changelog.cc ../master/masterconn.cc)
+add_library(metalogger STATIC ${METALOGGER_SOURCES} ../master/changelog.cc ../master/masterconn.cc)
 target_link_libraries(metalogger mfscommon)
 
 create_unittest(metalogger ${METALOGGER_TESTS})
diff --git a/src/metarestore/CMakeLists.txt b/src/metarestore/CMakeLists.txt
index 8adf9a8..3066568 100755
--- a/src/metarestore/CMakeLists.txt
+++ b/src/metarestore/CMakeLists.txt
@@ -12,7 +12,7 @@ else()
   file(GLOB METARESTORE_HSTRING_SOURCES ../master/hstring_memstorage.cc)
 endif()
 
-add_library(metarestore ${METARESTORE_SOURCES} ${METARESTORE_MASTER_SOURCES} ${METARESTORE_HSTRING_SOURCES}
+add_library(metarestore STATIC ${METARESTORE_SOURCES} ${METARESTORE_MASTER_SOURCES} ${METARESTORE_HSTRING_SOURCES}
   ../master/acl_storage.cc ../master/chunks.cc ../master/quota_database.cc ../master/chunk_goal_counters.cc
   ../master/restore.cc ../master/locks.cc ../master/task_manager.cc ../master/snapshot_task.cc
   ../master/setgoal_task.cc ../master/settrashtime_task.cc)
diff --git a/src/mount/CMakeLists.txt b/src/mount/CMakeLists.txt
index 99c758d..b466a07 100755
--- a/src/mount/CMakeLists.txt
+++ b/src/mount/CMakeLists.txt
@@ -19,6 +19,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 collect_sources(MOUNT)
 
 shared_add_library(mount ${MOUNT_SOURCES})
+set_target_properties(mount PROPERTIES OUTPUT_NAME "mfsmount")
 shared_target_link_libraries(mount mfscommon ${ADDITIONAL_LIBS})
 
 create_unittest(mount ${MOUNT_TESTS})
diff --git a/src/mount/client/CMakeLists.txt b/src/mount/client/CMakeLists.txt
index 5d1b251..9da97b8 100644
--- a/src/mount/client/CMakeLists.txt
+++ b/src/mount/client/CMakeLists.txt
@@ -1,26 +1,13 @@
 collect_sources(CLIENT)
 
 shared_add_library(lizardfs-client client.cc lizardfs_c_api.cc client_error_code.cc)
-shared_target_link_libraries(lizardfs-client mount)
+shared_target_link_libraries(lizardfs-client mount ${CMAKE_DL_LIBS})
 
 shared_add_library(lizardfs-client-cpp client.cc client_error_code.cc)
 shared_target_link_libraries(lizardfs-client-cpp mount)
 
-shared_target_link_libraries(lizardfs-client ${CMAKE_DL_LIBS})
-
-add_library(lizardfs-client_shared SHARED client.cc lizardfs_c_api.cc client_error_code.cc)
-set_target_properties(lizardfs-client_shared PROPERTIES OUTPUT_NAME "lizardfs-client")
-target_link_libraries(lizardfs-client_shared ${CMAKE_DL_LIBS} mount_pic)
-
-add_library(lizardfsmount_shared SHARED ${MOUNT_SOURCES} lizard_client_c_linkage.cc)
-target_link_libraries(lizardfsmount_shared mount_pic ${ADDITIONAL_LIBS})
-
-install(TARGETS lizardfs-client DESTINATION ${LIB_SUBDIR})
-install(TARGETS lizardfs-client_pic DESTINATION ${LIB_SUBDIR})
-install(TARGETS lizardfs-client-cpp DESTINATION ${LIB_SUBDIR})
-install(TARGETS lizardfs-client-cpp_pic DESTINATION ${LIB_SUBDIR})
-install(TARGETS lizardfsmount_shared DESTINATION ${LIB_SUBDIR})
-install(TARGETS lizardfs-client_shared DESTINATION ${LIB_SUBDIR})
+shared_add_library(lizardfsmount ${MOUNT_SOURCES} lizard_client_c_linkage.cc)
+shared_target_link_libraries(lizardfsmount mount ${ADDITIONAL_LIBS})
 
 install(FILES lizardfs_c_api.h DESTINATION ${INCL_SUBDIR})
 install(FILES ../../common/lizardfs_error_codes.h DESTINATION ${INCL_SUBDIR})
diff --git a/src/protocol/CMakeLists.txt b/src/protocol/CMakeLists.txt
index e6f1624..a28d010 100644
--- a/src/protocol/CMakeLists.txt
+++ b/src/protocol/CMakeLists.txt
@@ -18,7 +18,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 collect_sources(PROTOCOL)
 
-add_library(lzfsprotocol ${PROTOCOL_SOURCES})
+add_library(lzfsprotocol STATIC ${PROTOCOL_SOURCES})
 
 create_unittest(lzfsprotocol ${PROTOCOL_TESTS})
 link_unittest(lzfsprotocol mfscommon)
diff --git a/src/unittests/CMakeLists.txt b/src/unittests/CMakeLists.txt
index 8aa8eb9..fb24ce9 100644
--- a/src/unittests/CMakeLists.txt
+++ b/src/unittests/CMakeLists.txt
@@ -3,7 +3,7 @@ include_directories(${GTEST_INCLUDE_DIRS})
 add_subdirectory(mocks)
 
 collect_sources(UNITTESTS)
-add_library(unittests-helpers ${UNITTESTS_SOURCES})
+add_library(unittests-helpers STATIC ${UNITTESTS_SOURCES})
 
 if(APPLE)
   set(ENABLE_WHOLE_ARCHIVE "-Wl,-force_load")
diff --git a/src/unittests/mocks/CMakeLists.txt b/src/unittests/mocks/CMakeLists.txt
index cf9465d..3e64b78 100644
--- a/src/unittests/mocks/CMakeLists.txt
+++ b/src/unittests/mocks/CMakeLists.txt
@@ -1,6 +1,6 @@
 include_directories(${GTEST_INCLUDE_DIRS})
 
 collect_sources(UNITTESTMOCKS)
-add_library(unittest-mocks ${UNITTESTMOCKS_SOURCES})
+add_library(unittest-mocks STATIC ${UNITTESTMOCKS_SOURCES})
 create_unittest(unittest-mocks ${UNITTESTMOCKS_TESTS})
 link_unittest(unittest-mocks unittest-mocks mfscommon)
